ilastik/applets/dataSelection/dataSelectionGui.py

```diff
@@ -433,6 +434,21 @@ def getImageFileNamesToOpen(cls, parent_window, defaultDirectory):
         else:
             # otherwise, use native dialog of the present platform
             fileNames, _filter = QFileDialog.getOpenFileNames(parent_window, "Select Images", defaultDirectory, filt_all_str)
+        fileNames = DataSelectionGui.cleanFileList(fileNames)
+        return fileNames
+
+    @staticmethod
+    def cleanFileList(fileList: typing.List[str]) -> typing.List[str]:
+        fileNames = [pathlib.Path(selected_file) for selected_file in fileList]
+        # For the n5 extension the attributes.json file has to be selected in the file dialog.
+        # However we need just the *.n5 directory-file.
+        for i, fileName in enumerate(fileNames):
+            # On some OS's the open file dialog allows to return file names that do not exist
+            assert fileName.exists(), \
+                f"The file '{fileNames[i]}' does not exist."
+            if fileName.name.lower() == 'attributes.json' and '.n5' in fileName.as_posix():
```

Comment:
```suggestion
            if fileName.name.lower() == 'attributes.json' and fileName.parent.suffix == ".n5":
```

Reply:
nah, doesn't work for nested paths like `/somefile.n5/some/other/internal/path/attributes.json`

Reply:
`any(p.suffix == ".n5" for p in fileName.parents)` then? Otherwise, we could get false positives like `/spam/eggs.n5.not_really/ham` (not likely but still).

Reply:
I thought the same, thanks for this suggestion :)

Reply:
:+1: